packages/runtime/src/plugins/demand-control/calculateCost.ts

```diff
@@ -0,0 +1,198 @@
+import { getTypeInfo } from '@graphql-tools/delegate';
+import {
+  getDirective,
+  getDirectiveExtensions,
+  memoize3,
+} from '@graphql-tools/utils';
+import {
+  DocumentNode,
+  getArgumentValues,
+  getNamedType,
+  GraphQLOutputType,
+  GraphQLSchema,
+  isCompositeType,
+  isListType,
+  visit,
+  visitWithTypeInfo,
+} from 'graphql';
+
+type ListSizeAnnotation =
+  | {
+      slicingArguments: string[];
+    }
+  | {
+      assumedSize: number;
+    };
+
+type CostAnnotation = {
+  weight: number;
+};
+
+type DemandControlDirectives = {
+  cost?: CostAnnotation;
+  listSize?: ListSizeAnnotation;
+};
+
+function getDepthOfListType(type: GraphQLOutputType) {
+  let depth = 0;
+  while (isListType(type)) {
+    depth++;
+    type = type.ofType;
+  }
+  return depth;
+}
+
+export function createCalculateCost(defaultAssumedListSize?: number) {
+  return memoize3(function calculateCost(
+    schema: GraphQLSchema,
+    document: DocumentNode,
+    variables: Record<string, any>,
+  ) {
+    let cost = 0;
+    const factorQueue: number[] = [];
+    function timesFactor(c: number) {
+      for (const f of factorQueue) {
+        c *= f;
+      }
+      return c;
+    }
+    const typeInfo = getTypeInfo(schema);
+    visit(
+      document,
+      visitWithTypeInfo(typeInfo, {
+        OperationTypeDefinition(node) {
+          if (node.operation === 'mutation') {
+            cost += 10;
```

Comment:
should this be configurable?

Reply:
Hmm just implemented the same what Apollo does for mutations. This can be configurable still! Good idea!

Reply:
I added a new option for that! Thanks!